Lombiq.Tests.UI/MonkeyTesting/MonkeyTestingOptions.cs

```diff
@@ -48,7 +51,15 @@ public sealed class MonkeyTestingOptions
     {
         @"gremlins.species.clicker({
             log: true,
-            canClick: (element) => /https:\/\/localhost:\d\d\d\d.*/.test(element.href)})",
+            canClick: (element) => {
+                    for (; element && element !== document; element = element.parentNode) {
+                        if (!(/https:\/\/localhost:\d\d\d\d.*/.test(element.href) || !element.hasAttribute('href'))) {
+                            return false;
+                        }
+                    }
+
+                    return true;
+                }})",
```

Comment:
This causes a notable different execution compared to `dev` too, this one visits a lot more pages (e.g. in `TestAdminPagesAsMonkeyRecursivelyShouldWorkWithAdminUser`). This is not an issue per see but please investigate what's the reason behind it, since `dev` doesn't click on any non-localhost links either.

Reply:
What did you find?

Reply:
I'm still investigating, I just put it to "Done", because everything else is done.

Reply:
So the investigation is not done, the issue shouldn't be either :).

Reply:
Is there any evidence about it visiting significantly more pages? Because I ran the `dev` one and `SANTA-6` one, and it did the same thing. In fact, both of them ran between 1,5-2,5 mins. I think it's just pure luck, because it's random.

Reply:
I've watched the two running side by side a couple of times and this one always took longer with more pages. Running headless in the background always took longer too.

I tried again with a deliberate exception at the end, just to get screenshots for all pages, and obviously now I observe the exact opposite, `dev` visiting more pages. After multiple tries, I did get back to this visiting more (like 38 vs 104) but yeah, this seems to be just random. (Gremlin actions aren't supposed to be, but timings of the app responding vary and can cause this.)

```
    [Theory, Chrome]
    public Task TestAdminPagesAsMonkeyRecursivelyShouldWorkWithAdminUser(Browser browser) =>
        ExecuteTestAfterSetupAsync(
            async context =>
            {
                // Monkey tests needn't all start from the homepage. This one starts from the Orchard admin dashboard.
                await context.TestAdminAsMonkeyRecursivelyAsync(CreateMonkeyTestingOptions());
                throw new Exception("hehh");
            },
            browser,
            configuration =>
                // This is necessary to work around this bug: https://github.com/OrchardCMS/OrchardCore/issues/11420.
                configuration.AssertBrowserLog = messages => messages.ShouldNotContain(
                    message => IsValidAdminBrowserLogMessage(message),
                    messages.Where(IsValidAdminBrowserLogMessage).ToFormattedString()));
```